### PR TITLE
fix tests for mail 2.8

### LINKED ISSUE #1304: [master] 2 failing specs from actionmailer
bisected to https://github.com/mikel/mail/pull/607

this is a very old change (2013) that still isn't merged in mail 2.7.x
```
Failure:
BaseTest#test_can_pass_random_headers_in_as_a_hash_to_headers [/home/travis/build/mikel/mail/rails/actionmailer/test/base_test.rb:104]:
Expected: "1234@mikel.me.com"
  Actual: "<1234@mikel.me.com>"

Failure:
BaseTest#test_can_pass_random_headers_in_as_a_hash_to_mail [/home/travis/build/mikel/mail/rails/actionmailer/test/base_test.rb:96]:
Expected: "1234@mikel.me.com"
  Actual: "<1234@mikel.me.com>"
````

it doesn't look like a bug, but tests should be fixed

### PR DESCRIPTION
**Pull request by ahorek**
fixes https://github.com/mikel/mail/issues/1304

### CODE REVIEW

File: actionmailer/test/base_test.rb [hunk 1]
```diff
@@ -93,15 +93,15 @@ class BaseTest < ActiveSupport::TestCase
...
+    assert_match(/<?1234@mikel.me.com>?/, mail["In-Reply-To"].decoded)
   end
 
   test "can pass random headers in as a hash to headers" do
     hash = { "X-Special-Domain-Specific-Header" => "SecretValue",
             "In-Reply-To" => "1234@mikel.me.com" }
     mail = BaseMailer.welcome_with_headers(hash)
     assert_equal("SecretValue", mail["X-Special-Domain-Specific-Header"].decoded)
-    assert_equal("1234@mikel.me.com", mail["In-Reply-To"].decoded)
+    assert_match(/<?1234@mikel.me.com>?/, mail["In-Reply-To"].decoded)
```
Comment (jeremy) [inline actionmailer/test/base_test.rb]: Can we change to `assert_equal "<1234@mikel.me.com>"` rather than making the brackets optional?

Comment (ahorek) [reply actionmailer/test/base_test.rb]: it would fail on mail 2.7.1

**Review by jeremy — APPROVED** [review]
Looks good @ahorek. Would you add a Travis build step to test against mail master branch, like you did with Mail testing against Action Mailer master?

**Comment by ahorek** [comment]
I can't imagine a breaking change in Action Mailer that will break only Mail master. If that situation happens it should be probably fixed in the Mail gem that suppose to have very stable api between versions.
https://github.com/mikel/mail/pull/1303 will definitely help to catch failures earlier, but I don't think that a new stage in Rails will justify longer builds.
one of the goals of https://github.com/rails/rails/pull/34754 is to reduce the number of jobs and this goes against it, eventually we can do the same for other other dependencies like Sprockets, Puma, Redis etc...

**Comment by jeremy** [comment]
That's fair. In that case, let's use a different header in these tests whose decoded value is the same on Mail 2.7 and Mail 2.8. Or just remove the second In-Reply-To header since the first header demonstrates the behavior under test.

**Comment by ahorek** [comment]
this way it'll pass on both versions

**Review by jeremy — APPROVED** [review]
Thanks @ahorek

**Pull request merged**